Warning messages when compiling with all warnings enabled.

I am using this library with Rust. And when compiling the generated files I see warnings.

```text
warning: ts/toml/src/scanner.c: In function ‘tree_sitter_toml_external_scanner_destroy’:                                                                                     
warning: ts/toml/src/scanner.c:8:54: warning: unused parameter ‘payload’ [-Wunused-parameter]                                                                                
warning:     8 | void tree_sitter_toml_external_scanner_destroy(void *payload) {}
warning:       |                                                ~~~~~~^~~~~~~
...
...
warning: /usr/include/nodejs/deps/v8/include/v8.h:6199:16: warning: unused parameter ‘isolate’ [-Wunused-parameter]                                                          
warning:  6199 |       Isolate* isolate, Local<String> name) { 
```
It is more of an FYI, please close the issue if not relevant.

Thanks,

--- a/src/scanner.c
+++ b/src/scanner.c
@@ -1,5 +1,10 @@
 #include <tree_sitter/parser.h>
 
+#if defined(__GNUC__) || defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#endif
+
 enum TokenType {
   LINE_ENDING_OR_EOF
 };
@@ -33,3 +38,7 @@ bool tree_sitter_toml_external_scanner_scan(
 
   return false;
 }
+
+#if defined(__GNUC__) || defined(__clang__)
+#pragma GCC diagnostic pop
+#endif

